website/static/js/quickProjectSearchPlugin.js

```diff
@@ -0,0 +1,510 @@
+            contributors.data.forEach(function(contrib){
+                fullName = contrib.embeds.users.data.attributes.full_name;
+                contributorList.push(fullName);
+            });
+            self.contributorMapping[node.id] = contributorList;
+        };
+
+        // Gets contrib family name for display
+        self.getFamilyName = function(i, node) {
+            return node.embeds.contributors.data[i].embeds.users.data.attributes.family_name;
+        };
+
+        // Formats contrib family names for display
+        self.getContributors = function (node) {
+            var numContributors = node.embeds.contributors.links.meta.total;
+            if (numContributors === 1) {
+                return self.getFamilyName(0, node);
+            }
+            else if (numContributors === 2) {
+                return self.getFamilyName(0, node) + ' and ' +
+                    self.getFamilyName(1, node);
+            }
+            else if (numContributors === 3) {
+                return self.getFamilyName(0, node) + ', ' +
+                    self.getFamilyName(1, node) + ', and ' +
+                    self.getFamilyName(2, node);
+            }
+            else {
+                return self.getFamilyName(0, node) + ', ' +
+                    self.getFamilyName(1, node) + ', ' +
+                    self.getFamilyName(2, node) + ' + ' + (numContributors - 3);
+            }
+
+        };
+
+         // Formats date for display
+        self.formatDate = function (node) {
+            return new $osf.FormattableDate(node.attributes.date_modified).local;
+        };
+
+
+        self.sortAlphabeticalAscending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.title.toUpperCase();
+                var B = b.attributes.title.toUpperCase();
+                return (A < B) ? -1 : (A > B) ? 1 : 0;
+            });
+            self.sortState('alphaAsc');
+        };
+
+        self.sortAlphabeticalDescending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.title.toUpperCase();
+                var B = b.attributes.title.toUpperCase();
+                return (A > B) ? -1 : (A < B) ? 1 : 0;
+            });
+            self.sortState('alphaDesc');
+        };
+
+        self.sortDateAscending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.date_modified;
+                var B = b.attributes.date_modified;
+                return (A < B) ? -1 : (A > B) ? 1 : 0;
+            });
+            self.sortState('dateAsc');
+        };
+
+        self.sortDateDescending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.date_modified;
+                var B = b.attributes.date_modified;
+                return (A > B) ? -1 : (A < B) ? 1 : 0;
+            });
+            self.sortState('dateDesc');
+        };
```

Comment:
Seems like these sort functions could be refactored to duplicate less code. Not a blocker.